### ISSUE TITLE
How can someone submit releases to the gallery

### ISSUE THREAD

**Issue by horner** [utmapp/vm-downloads#5]
I built a rocky linux vm and I am curious how I could contribute it to the gallery.  I did not see pull requests on this project.

**Comment by jamesmortensen** [comment]
I also have a Debian 11 x86_64 VM I created (so I could run x86_64 Docker container images on the Mac M1).  I'm willing to share it.

**Comment by vleeuwenmenno** [comment]
> I also have a Debian 11 x86_64 VM I created (so I could run x86_64 Docker container images on the Mac M1). I'm willing to share it.

You really don't need to go all out with UTM to get x86_64 docker containers to work. I use them daily on my Mac mini M1 just with Docker desktop.

**Comment by jamesmortensen** [comment]
@vleeuwenmenno Thank you.  I wasn't aware Docker Desktop could run x86_64 container images.  Anyway, I moved away from Docker Desktop a few months back and run everything in UTM Debian VM's, one for ARM64 and one for x86_64.  I wrote about it [here on Codeluge](https://www.codeluge.com/post/setting-up-docker-on-macos-m1-arm64-to-use-debian-10.4-docker-engine/).  It's definitely easier working with Docker Desktop, but with this setup I'm only using open source and free components, and I've learned a lot more about how Docker works in the process.

**Comment by Lofter1** [comment]
I would also love to submit some. I would be more than willing to create and even maintain some security focused VMs.

**Comment by 3DCoded** [comment]
I made an app to store and share UTM templates. If you open an issue on the project, I can add it there. https://github.com/smattar2020/utm-templates. I will do my best to check it regularly.

**Comment by 3DCoded** [comment]
Or you could modify the code on my repo with a pull request. You just need a cloud service to upload the UTM zip's to and Xcode installed, then you can clone the repo, make changes, and then submit the pull request.

<img width="880" alt="Screen Shot 2022-04-26 at 4 53 57 PM" src="https://user-images.githubusercontent.com/69043333/165390619-9c9097db-1746-4548-8e5a-ad4225252a9f.png">

**Comment by 3DCoded** [comment]
Another option would be to use my code as a base then make your own repo for UTM templates.

**Comment by jamesmortensen** [comment]
Nice initiative, @smattar2020 I just installed it.  Why not speak with the maintainers about opening this up as a PR to the main UTM app.  It could not only connect to the official UTM gallery but also third party galleries as well.  This would avoid having to have a separate app just to download VM templates.  

Also, looking at the UTM gallery, the UTM project is just using GitHub releases page to host the VM's in ZIP archives.  I plan to use that myself, as long as the size doesn't exceed what GitHub allows.

**Comment by Lofter1** [comment]
Yeah. Seems like the "open in UTM" is also pretty easy to implement. I think an application to download templates is overkill (Thank you non the less @smatter2020).
We could try to start community maintained "repository" of images hosted on a website or something.

**Comment by jamesmortensen** [comment]
One possible idea is to create a Github Action in a repo that uploads zip files committed to a folder.  Users submit VMs by committing a utm.zip file in a specific folder and opening a PR. 
 When the PR is approved, the zip can then be uploaded as an artifact using https://github.com/actions/upload-artifact/ 

Then using something like readthedocs or another similar static site documentation generator, we could make that zip available on the website as user submitted, all without much manual intervention.

I think one thing that possibly is preventing UTM from taking user submitted VMs is that it is a manual process.  If it can be automated, then it reduces a lot of friction.

One concern though is how to make sure someone isn't uploading something nasty.  If I am not mistaken, rogue VMs can still do bad stuff to our host machines.

**Comment by 3DCoded** [comment]
OK. I will start the PR to the main UTM project and begin work on the OPEN IN UTM button. Maybe for when PR's are submitted to UTM templates, I can add a note to the action that warns that this VM has not been checked yet and to install it at your own risk. Thank you for the feedback (this is my first mac app I have made on GitHub).

**Comment by Lofter1** [comment]
> One concern though is how to make sure someone isn't uploading something nasty. If I am not mistaken, rogue VMs can still do bad stuff to our host machines.

That is a concern with almost every content submitted and maintained by the community, like arch linux's AUR. A warning that the content was not checked and can contain malicious software has to happen somewhere no matter what.

@smattar2020 the button is more for websites that serve as repositories, so you can download and import the VM image in one click to UTM.
If your app is being added to UTM, it would be more for browsing such a repository directly inside UTM, I think. So a "Open in UTM" button would not make much sense in there. (Let's hope it's being added, as something more basic like that exists in VM solutions like parallels and having a few pre-selected images to choose from when creating a new VM is a really neat feature already. Browsing a repository from the application would probably be a really amazing feature).

Anyway, I would be open to creating a repository for images on GitHub and writing the website to download and browse the images from. I would take something like homebrew taps (like the [homebrew core tap](https://github.com/Homebrew/homebrew-core)) as inspiration.

**Comment by 3DCoded** [comment]
As of now, I am not sure how to build UTM on my computer so I can't submit a PR.

**Comment by 3DCoded** [comment]
I also added VM size to UTM Templates. It's not in the release, but it is in a GitHub action. You can download it from https://github.com/smattar2020/utm-templates/actions/runs/2241288196 in the artifacts.

**Comment by 3DCoded** [comment]
Can we continue this in a discussion in utm-templates? It seems that this is becoming more about UTM Templates than UTM's vm-downloads. Here is the new discussion: https://github.com/smattar2020/utm-templates/discussions/7

**Comment by Lofter1** [comment]
So, I just finished writing the repository. It can be found [here](https://lofter1.github.io/UMT-VM-Gallery/). It's still very bare bones and I haven't uploaded an image I created myself yet (didn't have time yet), but it works.

**Comment by 3DCoded** [comment]
I embedded UTM Templates into UTM. My fork is available at https://github.com/smattar2020/UTM

**Comment by jamesmortensen** [comment]
@Lofter1 On Monday, I'll see if I can upload the Debian 11 AMD VM I have.

**Comment by jamesmortensen** [comment]
Debian 11 minimal amd64 is submitted via this PR - https://github.com/Lofter1/UMT-VM-Gallery/pull/2

**Comment by 3DCoded** [comment]
> Debian 11 minimal amd64 is submitted via this PR - [Lofter1/UMT-VM-Gallery#2](https://github.com/Lofter1/UMT-VM-Gallery/pull/2)

Can I use this in UTM Templates?

**Comment by Lofter1** [comment]
> Debian 11 minimal amd64 is submitted via this PR - https://github.com/Lofter1/UMT-VM-Gallery/pull/2

Awesome. Will look over it tomorrow. Have exams coming up, so it might take a bit.

**Comment by 3DCoded** [comment]
I am working on an alternative to UTM Templates, UTM Generator. I will link to it here when I have made a partially functional version to start with.
